src/Optimizer/Transformer/RewriteAmpUrls.php

```diff
@@ -268,9 +269,14 @@ private function adaptForSelfHosting(Document $document, $host, $errors)
             ! $this->configuration->get(RewriteAmpUrlsConfiguration::LTS)
         ) {
             try {
-                $urlParts = parse_url($host);
-                $origin   = "{$urlParts['scheme']}://{$urlParts['host']}";
-                $this->addMeta($document, 'runtime-host', $origin);
+                $url = new Url($host);
+
+                if (!empty($url->scheme) && !empty($url->host)) {
+                    $origin = "{$url->scheme}://{$url->host}";
+                    $this->addMeta($document, 'runtime-host', $origin);
```

Comment:
What about scheme-relative URLs? In that case, `$url->scheme` would be empty and the `$origin` would be `"//{$url->host}"`. I guess that's not supported, and we'd want an explicit HTTPS URL anyway.

Reply:
That would fail the first check (`!empty($url->scheme)`) and generate an error. Looks correct to me.

Reply:
I mean, if such URLs should be supported.

Reply:
Weston is right, we want an explicit https URL.